modules/feed-fetcher.js

```diff
@@ -21,9 +21,28 @@ export async function fetchFeed(feed, {allow_cached = false} = {}) {
         xhrPromise(request).catch(() => undefined),
         wait(DEFAULT_TIMEOUT),
     ]);
+
     if(!doc) {
-        console.error("failed to fetch", url);
-        return;
+        if (request.status == 200) {
+            // Retry the request but parsing the response by hand cleaning named entities.
+            let rqst = new XMLHttpRequest();
```

Comment:
Looks like one of my comments has been lost. Looks like in this case we have to download every feed having this kind of problem twice, which is not nice. Any ideas on making this work with just a single download?

Reply:
IIUC from [the spec](https://xhr.spec.whatwg.org/) and a few small tests, not setting `responseType` (or setting it to empty string) makes it possible to try reading `responseXML` first and in case of error read `responseText` instead.

Reply:
IIUC from [the spec](https://xhr.spec.whatwg.org/) and a few small tests, not setting `responseType` (or setting it to empty string) makes it possible to try reading `responseXML` first and in case of error read `responseText` instead.